src/content/1.5/code/agda/full.agda

```diff
@@ -0,0 +1,195 @@
+open import Relation.Binary.PropositionalEquality  using (_≡_)
+
+private variable
+  a b : Level
+  A B C C' : Set
+  f : A → B
+  g : B → A
+
+{- SECTION 5.1: INITIAL OBJECT ---------------------------------------------------
+The *initial object* is the object with exactly one morphism going from it to
+any other object in the category. It is unique up to isomorphism (when it exists).
+In the category of sets and functions, the initial object is the empty set.
+In Agda, we represent the empty set by the empty type ⊥.                         -}
+absurd : ⊥ → A
+
+{- `absurd` denotes a family of morphism from ⊥ to any object A in Set.
+Agda already defines this family and denotes it by `⊥-elim`
+(read "false elimination").                                                      -}
+absurd ()
+
+{- This is the familiar "ex falso quodlibet" principle: from a false premise
+anything follows.                                                                -}
+
+{- SECTION 5.2: TERMINAL OBJECT --------------------------------------------------}
+{- The *terminal object* is the object with exactly one morphism coming to it from
+any other object in the category. It is unique up to isomorphism (when it exists).
+In the category of sets and functions, the terminal object is the singleton.
+In Agda, we represent the singleton by the unit type ⊤.                          -}
+unit : A → ⊤
+unit _ = tt
+
+{- Here's another morphism (in Set) from any object to the two element set Bool: -}
+yes : A → Bool
+yes _ = true
+
+{- but Bool does not meet the uniqueness criterion of terminal object, as there
+are other morphisms from every set to Bool; e.g.,                                -}
+no : A → Bool
+no _ = false
+
+{- SECTION 5.4: ISOMORPHISMS -----------------------------------------------------}
+{- Morphism 𝑔 is the inverse of morphism 𝑓 if their composition is the identity
+morphism. These are actually two equations because there are two ways of composing
+two morphisms:                                                                   -}
+_ : f ∘ g ≡ id
+_ = {!!}   -- We leave an unfilled hole where a proof would normally go;
+           -- here a proof is impossible since we don't know anything about f or g.
+_ : g ∘ f ≡ id
+_ = {!!}
+
+{- SECTION 5.5: PRODUCTS ---------------------------------------------------------}
+{- The *product* of two objects `A` and `B` is an object `A × B` together with two
+morphisms `π₁ : A × B → A` and `π₂ : A × B → B` such that for any object `C`
+and morphisms `f : C → A`, `g : C → B` there is a unique morphism `h : C → A × B`
+such that `f ≡ π₁ ∘ h` and `g ≡ π₂ ∘ h`.                                         -}
+
+------------------------------------------------------------------------
+-- Syntax for Σ-types (see Data.Sum.Base of the Agda standard library)
+open import Agda.Builtin.Sigma hiding (module Σ)
+module Σ = Agda.Builtin.Sigma.Σ
+Σ-syntax = Σ
+syntax Σ-syntax A (λ x → B) = Σ[ x ∈ A ] B
+infix 2 Σ-syntax
```

Comment:
I just realise that explaining FP without dependent types (Σ and PI) is big limitation of this book :)